minimailgun/api.py

```diff
@@ -0,0 +1,96 @@
+
+import os
+import itertools
+import traceback
+from flask import Flask, Blueprint, jsonify, request, abort, current_app
+from werkzeug.exceptions import default_exceptions, HTTPException
+
+from minimailgun.store import store, MailLookupError, UnablToAddMessageError
+from minimailgun.tasks import handle_new_message
+
+
+mailgun_api = Blueprint('mini-mailgun', __name__)
+
+
+@mailgun_api.route('/mail', methods=['POST'])
+def create_mail():
+    if request.content_type != 'application/json':
+        abort(415, 'This api does not support content-type {mime}. Accepted mime-type: application/json'.format(
+            mime=request.mimetype))
+    if not request.accept_mimetypes.accept_json:
+        abort(406, 'This api only responds in application/json')
+    request_data = request.get_json()
+    if not request_data.get('from'):
+        abort(400, 'The from address is required.')
+    # other payload validation goes here.
```

Comment:
One thing I wanted to add was regex validation of email addresses being provided in from, to, cc etc.